serveradmin/serverdb/views.py

```diff
@@ -87,38 +100,35 @@ def history(request):
     if not object_id:
         raise Http404
 
-    adds = ChangeAdd.objects.filter(server_id=object_id)
-    updates = ChangeUpdate.objects.filter(server_id=object_id)
-    deletes = ChangeDelete.objects.filter(server_id=object_id).select_related()
+    # TODO: Change data model so that server_id is moved to ChangeCommit.
+    #       To me this would make most sense. If that is not possible (why?!)
+    #       write a proper (custom) query that performs.
+    where = {'server_id': object_id}
+    adds = ChangeAdd.objects.filter(**where)
+    updates = ChangeUpdate.objects.filter(**where)
+    deletes = ChangeDelete.objects.filter(**where)
 
     # @TODO Transform to PostgreSQL JSON and put a index here
     if search_string:
         adds = adds.filter(attributes_json__contains=search_string)
         updates = updates.filter(updates_json__contains=search_string)
         deletes = deletes.filter(attributes_json__contains=search_string)
 
-    adds = adds.select_related()
-    updates = updates.select_related()
-
     if commit_id:
         adds = adds.filter(commit__pk=commit_id)
         updates = updates.filter(commit__pk=commit_id)
         deletes = deletes.filter(commit__pk=commit_id)
 
+    related = ('commit__app', 'commit__user')
+    adds = adds.select_related(*related)
+    updates = updates.select_related(*related)
+    deletes = deletes.select_related(*related)
+
     change_list = ([('add', obj) for obj in adds] +
                    [('update', obj) for obj in updates] +
                    [('delete', obj) for obj in deletes])
     change_list.sort(key=lambda entry: entry[1].commit.change_on, reverse=True)
 
-    for change in change_list:
-        commit = change[1].commit
-        if commit.user:
-            commit.commit_by = commit.user.get_full_name()
```

Comment:
Is `commit_by` not needed anymore?

Reply:
the whole loop wasn't needed anymore because the template directly accesses the objects, which call `__str__`. this must be a relic